crates/bevy_core_widgets/src/core_slider.rs

```diff
@@ -0,0 +1,387 @@
+use core::ops::RangeInclusive;
+
+use accesskit::{Orientation, Role};
+use bevy_a11y::AccessibilityNode;
+use bevy_app::{App, Plugin};
+use bevy_ecs::event::Event;
+use bevy_ecs::lifecycle::OnInsert;
+use bevy_ecs::query::Has;
+use bevy_ecs::system::{In, ResMut};
+use bevy_ecs::world::DeferredWorld;
+use bevy_ecs::{
+    component::Component,
+    observer::Trigger,
+    query::With,
+    system::{Commands, Query, SystemId},
+};
+use bevy_input::keyboard::{KeyCode, KeyboardInput};
+use bevy_input::ButtonState;
+use bevy_input_focus::{FocusedInput, InputFocus, InputFocusVisible};
+use bevy_picking::events::{Drag, DragEnd, DragStart, Pointer, Press};
+use bevy_ui::{ComputedNode, ComputedNodeTarget, InteractionDisabled, UiGlobalTransform};
+
+/// Defines how the slider should behave when you click on the track (not the thumb).
+#[derive(Debug, Default)]
+pub enum TrackClick {
+    /// Clicking on the track lets you drag to edit the value, just like clicking on the thumb.
+    #[default]
+    Drag,
+    /// Clicking on the track increments or decrements the slider by [`StepSize`].
+    Step,
+    /// Clicking on the track snaps the value to the clicked position.
+    Snap,
+}
+
+/// A headless slider widget, which can be used to build custom sliders. Sliders have a value
+/// (represented by the [`SliderValue`] component) and a range (represented by [`SliderRange`]).
+/// An optional step size can be specified via [`SliderStep`].
+///
+/// Typically a slider will contain entities representing the "track" and "thumb" elements.
+/// The core slider makes no assumptions about the hierarchical structure of these elements,
+/// but expects that the thumb will be marked with a [`CoreSliderThumb`] component.
+#[derive(Component, Debug, Default)]
+#[require(
+    AccessibilityNode(accesskit::Node::new(Role::Slider)),
+    CoreSliderDragState,
+    SliderValue,
+    SliderRange,
+    SliderStep
+)]
+pub struct CoreSlider {
+    /// This value represents the size, in pixels, that the slider's track length should be reduced
+    /// (during picking calculations) to account for the fact that the slider thumb takes up space.
+    /// This can be different than the actual visible size of the thumb.
+    pub thumb_size: f32,
```

Comment:
IMO `thumb_size` should be inferred from the size of the `CoreSliderThumb` node. If you want the thumb to not match the thumb size, then you can add a child of the `CoreSliderThumb` with whatever visual effect you want for it.

Setting `thumb_size` in the `CoreSlider` component like this means user code will be more fragile and unwieldy, since users will need to manually synchronize `thumb_size` with the actual node size (if they want them synced).

Reply:
Agreed. If we want to allow for manual tuning beyond that, it should be in the form of a multiplier that defaults to 1.0.